servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/gc/AbstractRestGCTest.java

```diff
@@ -0,0 +1,416 @@
+        commitSingleOp(
+            prefix, branch2, branch2.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    fillExpectedContents(branch1, 1, expectedExpired);
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    Instant defaultCutoffTime = Instant.now();
+
+    // commits for "table_1"
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 45, "cid_1", "table_1", "file4", table1.content, null);
+    fillExpectedContents(branch1, 2, expectedLive);
+
+    // commits for "table_2"
+    table2 =
+        commitSingleOp(
+            prefix, branch2, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    Instant branch2CutoffTime = Instant.now();
+    Map<String, Instant> perRefCutoffTime = new HashMap<>();
+    perRefCutoffTime.put(branch2.getName(), branch2CutoffTime);
+
+    // commits for "table_2"
+    table2 =
+        commitSingleOp(
+            prefix, branch2, table2.hash, 45, "cid_2", "table_2", "file4", table2.content, null);
+    fillExpectedContents(branch2, 1, expectedLive);
+
+    // table1 -> one expired and two live commits.
+    // table 2 -> two expired and one live commit.
+    performGc(
+        defaultCutoffTime,
+        perRefCutoffTime,
+        expectedLive,
+        expectedExpired,
+        new HashSet<>(Arrays.asList(prefix + "_cid_1", prefix + "_cid_2")));
+  }
+
+  @Test
+  public void testMultiRefAssignRef() throws BaseNessieClientServerException {
+    String prefix = "test7";
+    Map<String, Set<Content>> expectedExpired = new HashMap<>();
+    Map<String, Set<Content>> expectedLive = new HashMap<>();
+    Branch before = createBranch("multiRefAssignRef_0");
+    Branch branch1 = createBranch("multiRefAssignRef_1");
+    Branch branch2 = createBranch("multiRefAssignRef_2");
+    // commit for "table_1" on branch1
+    CommitOutput table1 =
+        commitSingleOp(
+            prefix, branch1, branch1.getHash(), 42, "cid_1", "table_1", "file1", null, null);
+    // commit for "table_2" on branch2
+    CommitOutput table2 =
+        commitSingleOp(
+            prefix, branch2, branch2.getHash(), 42, "cid_2", "table_2", "file1", null, null);
+    fillExpectedContents(branch1, 1, expectedExpired);
+    fillExpectedContents(branch2, 1, expectedExpired);
+
+    final Instant cutoffTime = Instant.now();
+
+    // commits for "table_1"
+    table1 =
+        commitSingleOp(
+            prefix, branch1, table1.hash, 44, "cid_1", "table_1", "file3", table1.content, null);
+    fillExpectedContents(branch1, 1, expectedLive);
+
+    // commits for "table_2"
+    table2 =
+        commitSingleOp(
+            prefix, branch2, table2.hash, 44, "cid_2", "table_2", "file3", table2.content, null);
+    fillExpectedContents(branch2, 1, expectedExpired);
```

Comment:
Why is this expired? It's committed after cut-off-time and branch2 is still alive?

Reply:
because it is assign reference (dead commit). Line 401